subsys/llext/llext_load.c

```diff
@@ -625,20 +643,14 @@ int do_llext_load(struct llext_loader *ldr, struct llext *ext,
 	}
 
 out:
-	llext_free(ldr->sect_map);
+	llext_load_cleanup(ldr, ext, ret);
```

Comment:
I'm not too fond of passing the ret into a function that does branching on it. I'd rather see the branch at this level myself. Something like...

```c 
llext_free_loader(ldr, ext);
if (ret != 0) {
   LOG_ERR("Failed to load extension...");
   llext_free_extension(ext);
} else {
  LOG_DBG("Loaded extension ...");
}
return ret;
```

Reply:
I see your point. Will merge the code in llext_load_cleanup after the label.